torch/distributions/categorical.py

```diff
@@ -115,7 +116,9 @@ def log_prob(self, value):
         return log_pmf.gather(-1, value).squeeze(-1)
 
     def entropy(self):
-        p_log_p = self.logits * self.probs
+        min_real = torch.finfo(self.logits.dtype).min
+        logits = torch.clamp(self.logits, min=min_real)
+        p_log_p = logits * self.probs
```

Comment:
This will be a good place to use [xlogy](https://github.com/pytorch/pytorch/issues/22656) when available.

Reply:
I think xlogy calculates `x * log(y)` where here we want `x * y`

Reply:
That's right, we are using the precomputed probs / logits attributes directly.